File: lib/src/models/destiny_historical_weapon_stats_data.dart
import 'package:json_annotation/json_annotation.dart';

import 'destiny_historical_weapon_stats.dart';

part 'destiny_historical_weapon_stats_data.g.dart';

@JsonSerializable()
class DestinyHistoricalWeaponStatsData{
	
	DestinyHistoricalWeaponStatsData();

	factory DestinyHistoricalWeaponStatsData.fromJson(Map<String, dynamic> json) => _$DestinyHistoricalWeaponStatsDataFromJson(json);

	/// List of weapons and their perspective values.
	@JsonKey(name:'weapons')
	List<DestinyHistoricalWeaponStats> weapons;

	
	
	Map<String, dynamic> toJson() => _$DestinyHistoricalWeaponStatsDataToJson(this);
}

File: lib/src/models/DestinyHistoricalWeaponStatsData.cs
using System.Runtime.Serialization;
using System.Collections.Generic;

namespace BungieNetApi.Model {
	[DataContract]
	public class DestinyHistoricalWeaponStatsData{

		/// <summary>
		/// List of weapons and their perspective values.
		/// </summary>
		[DataMember(Name="weapons", EmitDefaultValue=false)]
		public List<DestinyHistoricalWeaponStats> Weapons { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as DestinyHistoricalWeaponStatsData);
        }

		public bool Equals(DestinyHistoricalWeaponStatsData input)
		{
			if (input == null) return false;

			return
				(
                    Weapons == input.Weapons ||
                    (Weapons != null && Weapons.Equals(input.Weapons))
                ) ;
		}

		/* 
		public override int GetHashCode()
		{
			unchecked // Overflow is fine, just wrap
			{
				int hashCode = 41;
				hashCode = hashCode * 59 + this.Weapons.GetHashCode();
				return hashCode;
			}
		}*/
	}
}

